patches/chromium/fix_crash_loading_non-standard_schemes_in_iframes.patch

```diff
@@ -36,11 +36,31 @@ index e91fb6b203b55d4937e9f4b4ed8fd5e8efb5aa10..855cb90260de04b90b8bbf4a8733e086
    return origin_with_debug_info;
  }
  
+diff --git a/content/browser/renderer_host/render_frame_host_impl.cc b/content/browser/renderer_host/render_frame_host_impl.cc
+index 8784a4a93993110879564c2cd5520e2d69f59b07..a2d6831d94623349c1ea7b7bcd53383c142c6c34 100644
+--- a/content/browser/renderer_host/render_frame_host_impl.cc
++++ b/content/browser/renderer_host/render_frame_host_impl.cc
+@@ -10153,9 +10153,14 @@ void RenderFrameHostImpl::CommitNavigation(
+       ProcessLock::FromSiteInfo(GetSiteInstance()->GetSiteInfo());
+   auto browser_calc_origin_to_commit =
+       navigation_request->GetOriginToCommitWithDebugInfo();
++  url::Origin origin_to_check = browser_calc_origin_to_commit.first.value();
++  if(!common_params->url.IsStandard()) {
++    //Custom URLS fail the CanAccessDataForOrigin check with browser_calc_origin_to_commit.first.value
++    origin_to_check = url::Origin::Create(common_params->url);
++  }
+   if (!process_lock.is_error_page() && !is_mhtml_subframe &&
+       !policy->CanAccessDataForOrigin(
```

Comment:
Shouldn't this be swapped for `CanCommitOriginAndUrl` given the value `browser_calc_origin_to_commit` is obtained from `GetOriginForURLLoaderFactoryAfterResponseWithDebugInfo` which already swaps the policy API.

Reply:
@deepak1556 see the patch explanation - it describes why we can't use `CanCommitOriginAndUrl`

Reply:
I don't see a mention of it in the patch description, also why can't we use `CanCommitOriginAndUrl` since the values being checked are same as the one in `NavigationRequest::GetOriginForURLLoaderFactoryAfterResponseWithDebugInfo`

Reply:
https://github.com/electron/electron/pull/39774/commits/38fef075fc5690f7db6d4bbcabbe877a1618a964

Reply:
I was suggesting to use `RenderFrameHost::CanCommitOriginAndUrl` to match the check in `NavigationRequest::GetOriginForURLLoaderFactoryAfterResponseWithDebugInfo`, given we have access to the navigation request, we should have all the parameters needed to make the same check. `ChildProcessSecurityPolicy::CanCommitOriginAndUrl` has a different implementation, it will still end up checking `CanAccessDataForOrigin` that will lead to the crash.

Reply:
Refactored in https://github.com/electron/electron/pull/39879